Sources_v3/StreamChatUI/ChatChannel/MessageComposer/MessageComposerDocumentAttachmentView.swift

```diff
@@ -0,0 +1,77 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class MessageComposerDocumentAttachmentView<ExtraData: ExtraDataTypes>: ChatMessageAttachmentInfoView<ExtraData> {
+    // MARK: - Properties
+    
+    public var discardButtonHandler: (() -> Void)?
+    
+    // MARK: - Subviews
+
+    public private(set) lazy var fileIconImageView: UIImageView = {
+        let imageView = UIImageView().withoutAutoresizingMaskConstraints
+        imageView.contentMode = .center
+        return imageView
+    }()
+
+    // MARK: - Overrides
+    
+    override open func setUp() {
+        super.setUp()
+        
+        let tapRecognizer = UITapGestureRecognizer(target: self, action: #selector(discard))
+        addGestureRecognizer(tapRecognizer)
+    }
+
+    override public func defaultAppearance() {
+        backgroundColor = .white
+        layer.cornerRadius = 15
+        layer.masksToBounds = true
+        layer.borderWidth = 1
+        layer.borderColor = uiConfig.colorPalette.incomingMessageBubbleBorder.cgColor
+        
+        fileSizeLabel.textColor = uiConfig.colorPalette.subtitleText
+        fileNameLabel.textColor = uiConfig.colorPalette.text
+        
+        actionIconImageView.image = UIImage(named: "discardAttachment", in: .streamChatUI)
+    }
+
+    override open func setUpLayout() {
+        addSubview(fileIconImageView)
+        addSubview(actionIconImageView)
+        addSubview(fileNameAndSizeStack)
+
+        NSLayoutConstraint.activate([
+            fileIconImageView.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor),
+            fileIconImageView.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor),
+            fileIconImageView.bottomAnchor.constraint(equalTo: layoutMarginsGuide.bottomAnchor),
+            
+            actionIconImageView.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor),
+            actionIconImageView.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor),
+            actionIconImageView.leadingAnchor.constraint(
+                equalToSystemSpacingAfter: fileNameAndSizeStack.trailingAnchor,
+                multiplier: 1
+            ),
+            
+            fileNameAndSizeStack.leadingAnchor.constraint(
+                equalToSystemSpacingAfter: fileIconImageView.trailingAnchor,
+                multiplier: 2
```

Comment:
This is an interesting workaround 🤔 Maybe that's the way how we should do layouts without magic numbers? cc @ManWithBear

Reply:
We actually using this approach in `MessageList` tree.
Can't say that it's better / worse, Instead of magic `16` you getting magic `2`. In the end it's all the same

Reply:
@ManWithBear But the magic "2" here just says - this margin should be twice that margin. There will always be magic numbers I just think we should not use them if not needed.